src/Cafe/OS/libs/nsyshid/nsyshid.cpp

```diff
@@ -759,6 +762,19 @@ namespace nsyshid
 		return nullptr;
 	}
 
+	bool Backend::FindDeviceById(uint16 vendorId, uint16 productId)
+	{
+		std::lock_guard<std::recursive_mutex> lock(this->m_devicesMutex);
+		for (const auto& device : m_devices)
+		{
+			if (device->m_vendorId == vendorId && device->m_productId == productId)
+			{
+				return true;
+			}
+		}
+		return false;
+	}
```

Comment:
This won't work to check for globally connected devices, as we just walk through the backend's own list of devices that should be detached when the backend is detached.
Although I generally don't like the word "global", maybe we could rename this method to something like `FindGlobalDeviceById` and make it iterate over the `deviceList` and lock the `hidMutex` that are part of `nsyshid`?
Suggestions welcome :sweat_smile:

Reply:
Have kept the same name, but just have a backend version and an nsyshid version, let me know if that looks right!